dbms/src/Functions/IFunction.h

```diff
@@ -28,6 +28,8 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+//#pragma GCC diagnostic push
```

Comment:
@alexey-milovidov 
dont like that I dodnt pop the pragma. It means, anything including this header gets the error hidden. Any idea how to solve it?

Reply:
We should rename that function.

Reply:
Looks like I will be busy until the end of September. Hopefully will be beck to finish this PR

Reply:
Ok, we'll wait.